basenode/src/main/java/io/coti/basenode/services/BaseNodeFeesService.java

```diff
@@ -1,126 +1,140 @@
+            if (tokenFeeData == null) {
+                throw new CotiRunTimeException(String.format("no matching token found to calculate fee of type: %s", nodeFeeType.name()));
+            }
         }
 
-        return feeAmount;
+        return tokenFeeData.getFeeAmount(amount);
+    }
+
+    @Override
+    public ResponseEntity<IResponse> setFeeValue(@Valid RatioTokenFeeSetRequest ratioTokenFeeSetRequest) {
+
+        FeeData feeData = new FeeData(ratioTokenFeeSetRequest.getFeePercentage(), ratioTokenFeeSetRequest.getFeeMinimum(),
+                ratioTokenFeeSetRequest.getFeeMaximum());
+        RatioTokenFeeData ratioTokenFeeData = new RatioTokenFeeData(ratioTokenFeeSetRequest.getTokenSymbol(),
+                ratioTokenFeeSetRequest.getNodeFeeType(), feeData);
+        Hash tokenHash = OriginatorCurrencyCrypto.calculateHash(ratioTokenFeeSetRequest.getTokenSymbol());
+        if (ratioTokenFeeData.valid()) {
+            TokenFeeData nodeFeeData = getTokenFeeData(tokenHash, ratioTokenFeeSetRequest.getNodeFeeType());
+            if (nodeFeeData instanceof ConstantTokenFeeData) {
+                nodeFees.deleteByHash(nodeFeeData.getHash());
+            }
+            nodeFees.put(ratioTokenFeeData);
+            GetNodeFeesDataResponse getNodeFeesDataResponse = new GetNodeFeesDataResponse();
+            getNodeFeesDataResponse.setNodeFeeDataArrayList(Collections.singletonList(ratioTokenFeeData));
+
+            return ResponseEntity.status(HttpStatus.OK).body(getNodeFeesDataResponse);
+
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new Response(INVALID_PARAMETERS_MESSAGE, STATUS_ERROR));
     }
 
-    private boolean newFeeValuesValid(NodeFeeSetRequest nodeFeeSetRequest) {
-        return !(nodeFeeSetRequest.getFeePercentage().compareTo(BigDecimal.ZERO) <= 0 ||
-                nodeFeeSetRequest.getFeePercentage().compareTo(BigDecimal.valueOf(100)) > 0 ||
-                nodeFeeSetRequest.getFeeMinimum().compareTo(BigDecimal.ZERO) <= 0 ||
-                nodeFeeSetRequest.getFeeMaximum().compareTo(nodeFeeSetRequest.getFeeMinimum()) < 0);
+    @Override
+    public ResponseEntity<IResponse> deleteFeeValue(DeleteTokenFeeRequest deleteTokenFeeRequest) {
+
+        Hash tokenHash = OriginatorCurrencyCrypto.calculateHash(deleteTokenFeeRequest.getTokenSymbol());
+        TokenFeeData nodeFeeData = getTokenFeeData(tokenHash, deleteTokenFeeRequest.getNodeFeeType());
+        if (nodeFeeData != null) {
+            nodeFees.deleteByHash(nodeFeeData.getHash());
+            GetNodeFeesDataResponse getNodeFeesDataResponse = new GetNodeFeesDataResponse();
+            getNodeFeesDataResponse.setNodeFeeDataArrayList(Collections.singletonList(nodeFeeData));
+
+            return ResponseEntity.status(HttpStatus.OK).body(getNodeFeesDataResponse);
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new Response(INVALID_PARAMETERS_MESSAGE, STATUS_ERROR));
```

Comment:
Response error message is not reflect actual reason of error, should be like NodeFeeData was not found or something like that.